storage/internal/benchmarks/utils.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"log"
+	"math/rand"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+)
+
+const (
+	alphabet                   = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	lowercaseLettersAndNumbers = "abcdefghijklmnopqrstuvwxyz0123456789"
+	uppercaseLetters           = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	asciiChars                 = " !\"#$%&\\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+)
+
+// returns a value in range [min, max]
+// includes endpoints in possible values to return
+func randomValue(min, max int64) int64 {
+	return rand.Int63n(max-min+1) + min
+}
+
+func randomBucketName(prefix string) string {
+	var sb strings.Builder
+	// The total length of the bucket name must be <= 63 characters
+	maxLen := 63
+	date := time.Now().Format("06-01-02-1504")
+
+	sb.WriteString(prefix)
+	sb.WriteRune('-')
+	sb.WriteString(date)
+	sb.WriteRune('_')
+
+	maxRandomChars := maxLen - sb.Len()
+	sb.WriteString(randomString(maxRandomChars, lowercaseLettersAndNumbers))
+
+	return sb.String()
+}
+
+func randomObjectName() string {
+	// GCS accepts object name up to 1024 characters, but 128 seems long enough to
+	// avoid collisions.
+	maxLen := 128
+
+	return randomString(maxLen, lowercaseLettersAndNumbers+uppercaseLetters)
+}
+
+// random string between 1 and maxLen
```

Comment:
Is it problematic memory-wise to use this for larger objects? Also, to compare with other languages, I'm wondering if measuring the whole round trip from disk would be preferable.

Reply:
Memory-wise, this is trying to emulate a buffer of "AppSizeBuffer". I'm not sure if other languages are getting from disk. My understanding of the C++ implementation is that it does something similar